src/components/surveys/survey-design/question-edit/rhs-subcontrols/Year.tsx

```diff
@@ -110,54 +40,56 @@ const Year: React.FunctionComponent<{
     if (range.max && range.min && range.max < range.min) {
       return 'RANGE'
     }
-    if (gtCurrent && !allowFuture) {
+    if (gtCurrent && allowFuture !== undefined && !allowFuture) {
       return 'NO_FUTURE_YEARS'
     }
-    if (ltCurrent && !allowPast) {
+    if (ltCurrent && allowPast !== undefined && !allowPast) {
       return 'NO_PAST_YEARS'
     }
 
     return null
   }
 
-  React.useEffect(() => {
-    const error = validate(range || {}, allowPast, allowFuture)
+  const error: keyof typeof ErrorMessages | null = validate(range, allowPast, allowFuture)
 
-    setError(error)
+  React.useEffect(() => {
     if (!error) {
-      onUpdateFormat({
-        ...step.inputItem.formatOptions,
+      const formatOptions = {
         minimumYear: range?.min,
         maximumYear: range?.max,
-        allowFuture,
-        allowPast,
-      })
+        allowFuture: allowFuture,
+        allowPast: allowPast,
+      }
+      // only call onChange if the formatOptions have actually changed
+      if (!_.isEqual(formatOptions, prevFormatOptions)) {
+        setPrevFormatOptions(formatOptions)
+        const inputItem = {...step.inputItem, formatOptions: formatOptions}
+        onChange({...step, inputItem})
+      }
     }
-  }, [range, allowPast, allowFuture, onUpdateFormat, step.inputItem])
+  }, [range, allowPast, allowFuture, prevFormatOptions, error, onChange, step])
```

Comment:
If you make this fully controlled, you could probably call `validate` and `onChange` in the `YearRadioGroup`'s `onChange` prop and then completely remove this effect (and not have to worry about an entire class of bugs)